elliottlib/attach_cve_flaws.py

```diff
@@ -37,24 +34,37 @@ def get_corresponding_flaw_bugs(bzapi, tracker_bugs):
 
 
 def is_first_fix(bzapi, flaw_bug, current_target_release, tracker_ids_to_be_ignored=[]):
-    other_flaw_trackers = bzapi.query(bzapi.build_query(
+    """
+    Check if a flaw bug is considered a first-fix for a target release
+    """
+    # get all the tracker bugs for a flaw bug
+    # but only for OCP product
+    tracker_ids = [t for t in flaw_bug.depends_on if t not in tracker_ids_to_be_ignored]
+    tracker_bugs = bzapi.query(bzapi.build_query(
         product='OpenShift Container Platform',
-        bug_id=[t for t in flaw_bug.depends_on if t not in tracker_ids_to_be_ignored],
+        bug_id=tracker_ids,
     ))
 
-    def _filter_tracker(bug):
-        current_major_version = util.minor_version_tuple(current_target_release[0])[0]
+    def same_major_release(bug):
+        current_major_version = util.minor_version_tuple(current_target_release)[0]
         bug_target_major_version = util.minor_version_tuple(bug.target_release[0])[0]
         return bug_target_major_version == current_major_version
 
-    def _already_fixed(bug):
+    def already_fixed(bug):
         if bug.status == 'RELEASE_PENDING':
             return True
         if bug.status == 'CLOSED' and bug.resolution in ['ERRATA', 'CURRENTRELEASE', 'NEXTRELEASE']:
             return True
         return False
 
-    return not any([_already_fixed(t) for t in filter(_filter_tracker, other_flaw_trackers)])
+    # if any tracker bug for the flaw bug
```

Comment:
I refactored this to make it **very clear** of what was happening for future developers working on this :)

Reply:
Need to mention components here. CVEs can affect multiple components, we need to check if a tracker represents a first fix for the given component, not the general CVE.

Reply:
Thanks, since this is something new I'll write up a story for this

Reply:
@sfowl Please review https://issues.redhat.com/browse/ART-2725